esmvalcore/cmor/_fixes/cmip5/bcc_csm1_1.py

```diff
@@ -1,9 +1,65 @@
 """Fixes for bcc-csm1-1."""
+import iris
 import numpy as np
 from scipy.interpolate import InterpolatedUnivariateSpline
 from scipy.ndimage import map_coordinates
 
 from ..fix import Fix
+from ..shared import fix_bounds
+
+
+class Cl(Fix):
+    """Fixes for cl."""
+
+    def fix_metadata(self, cubes):
+        """Fix hybrid sigma pressure coordinate.
+
+        Parameters
+        ----------
+        cubes : iris.cube.CubeList
+            Input cubes which need to be fixed.
+
+        Returns
+        -------
+        iris.cube.CubeList
+
+        """
+        cl_cube = self.get_cube_from_list(cubes)
+
+        # Remove all existing aux_factories
+        for aux_factory in cl_cube.aux_factories:
+            cl_cube.remove_aux_factory(aux_factory)
+
+        # Fix bounds
+        coords_to_fix = ['b']
+        try:
+            cl_cube.coord(var_name='a')
+            coords_to_fix.append('a')
+        except iris.exceptions.CoordinateNotFoundError:
+            coords_to_fix.append('ap')
+        fix_bounds(cl_cube, cubes, coords_to_fix)
+
+        # Fix bounds for ap if only a is given in original file
+        ap_coord = cl_cube.coord(var_name='ap')
+        if ap_coord.bounds is None:
+            cl_cube.remove_coord(ap_coord)
+            a_coord = cl_cube.coord(var_name='a')
+            p0_coord = cl_cube.coord(var_name='p0')
+            ap_coord = a_coord * p0_coord.points[0]
+            ap_coord.units = a_coord.units * p0_coord.units
+            ap_coord.rename('vertical pressure')
+            ap_coord.var_name = 'ap'
+            cl_cube.add_aux_coord(ap_coord, cl_cube.coord_dims(a_coord))
+
```

Comment:
```suggestion
        # Fix bounds
        try:
            a_coord = cl_cube.coord(var_name='a')
            # ap_coord was constructed by iris without bounds
            # rebuild ap_coord
            cl_cube.remove_coord(ap_coord)
            fix_bounds(cl_cube, cubes, ['b', 'a'])
            p0_coord = cl_cube.coord(var_name='p0')
            ap_coord = a_coord * p0_coord.points[0]
            ap_coord.units = a_coord.units * p0_coord.units
            ap_coord.rename('vertical pressure')
            ap_coord.var_name = 'ap'
            cl_cube.add_aux_coord(ap_coord, cl_cube.coord_dims(a_coord))
        except iris.exceptions.CoordinateNotFoundError:
            coords_to_fix.append('ap')
            fix_bounds(cl_cube, cubes, ['b', 'ap'])
            ap_coord = cl_cube.coord(var_name='ap')

```

Reply:
Unfortunately that won't work, since `fix_bounds(cl_cube, cubes, ['b', 'ap'])` will look if an `ap_bounds` variable is in `cubes`, which is not the case if the file has `a`/`p0` coordinates. Thus, it will raise an exception.

I will add your comment `# ap_coord was constructed by iris without bounds rebuild ap_coord` to the code, that will make it clearer.